src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -101,20 +103,30 @@ private OverdueFineCalculatorService(Clients clients) {
 
     OverdueFinePolicy overdueFinePolicy = loan.getOverdueFinePolicy();
     if (overdueMinutes > 0 && overdueFinePolicy != null) {
-      Double maxFine = loan.wasDueDateChangedByRecall() ?
-        overdueFinePolicy.getMaxOverdueRecallFine() :
-        overdueFinePolicy.getMaxOverdueFine();
-
-      OverdueFineInterval interval = overdueFinePolicy.getOverdueFineInterval();
-      Double overdueFinePerInterval = loan.getOverdueFinePolicy().getOverdueFine();
-
-      if (maxFine != null && interval != null && overdueFinePerInterval != null) {
-        double numberOfIntervals = Math.ceil(overdueMinutes.doubleValue() /
-          interval.getMinutes().doubleValue());
-        overdueFine = overdueFinePerInterval * numberOfIntervals;
-
-        if (maxFine > 0) {
-          overdueFine = Math.min(overdueFine, maxFine);
+      OverdueFinePolicyFineInfo fineInfo = overdueFinePolicy.getFineInfo();
+      OverdueFinePolicyLimitInfo limitInfo = overdueFinePolicy.getLimitInfo();
+      if (fineInfo != null && limitInfo != null) {
+        Double finePerInterval = loan.wasDueDateChangedByRecall() ?
+          fineInfo.getOverdueRecallFine() :
+          fineInfo.getOverdueFine();
+
+        OverdueFineInterval interval = loan.wasDueDateChangedByRecall() ?
+          fineInfo.getOverdueRecallFineInterval() :
+          fineInfo.getOverdueFineInterval();
```

Comment:
I'm curious, why is this ternary operation formatted like this rather than something like below?
```suggestion
        OverdueFineInterval interval = loan.wasDueDateChangedByRecall() 
          ? fineInfo.getOverdueRecallFineInterval() 
          : fineInfo.getOverdueFineInterval();
```

Reply:
Well, bad old habits sometimes kick in. This has been removed (OverdueFinePolicy provides these params now).